spec/unit/util/windows/registry_spec.rb

```diff
@@ -71,70 +71,94 @@ class TestRegistry
 
       expect(subject.values(key)).to eq({})
     end
 
-    context "when reading non-ASCII values" do
-      # registered trademark symbol
-      let(:data) do
-        str = [0xAE].pack("C")
-        str.force_encoding('US-ASCII')
-        str
-      end
-
-      def expects_registry_value(array)
-        key.expects(:each_value).multiple_yields(array)
-
-        subject.values(key).first[1]
-      end
-
-      # The win32console gem applies this regex to strip out ANSI escape
-      # sequences. If the registered trademark had encoding US-ASCII,
-      # the regex would fail with 'invalid byte sequence in US-ASCII'
-      def strip_ansi_escapes(value)
-        value.sub(/([^\e]*)?\e([\[\(])([0-9\;\=]*)([a-zA-Z@])(.*)/, '\5')
-      end
-
-      it "encodes REG_SZ according to the current code page" do
-        reg_value = ['string', Win32::Registry::REG_SZ, data]
-
-        value = expects_registry_value(reg_value)
+    it "passes REG_DWORD through" do
+      reg_value = ['dword', Win32::Registry::REG_DWORD, '1']
 
-        strip_ansi_escapes(value)
-      end
-
-      it "encodes REG_EXPAND_SZ based on the current code page" do
-        reg_value = ['expand', Win32::Registry::REG_EXPAND_SZ, "%SYSTEMROOT%\\#{data}"]
-
-        value = expects_registry_value(reg_value)
-
-        strip_ansi_escapes(value)
-      end
-
-      it "encodes REG_MULTI_SZ based on the current code page" do
-        reg_value = ['multi', Win32::Registry::REG_MULTI_SZ, ["one#{data}", "two#{data}"]]
+      value = expects_registry_value(reg_value)
 
-        value = expects_registry_value(reg_value)
+      expect(Integer(value)).to eq(1)
+    end
 
-        value.each { |str| strip_ansi_escapes(str) }
+    context "when reading non-ASCII values" do
+      ENDASH_UTF_8 = [0xE2, 0x80, 0x93]
+      ENDASH_UTF_16 = [0x2013]
+      TM_UTF_8 = [0xE2, 0x84, 0xA2]
+      TM_UTF_16 = [0x2122]
+
+      let (:hklm) { Win32::Registry::HKEY_LOCAL_MACHINE }
+      let (:puppet_key) { "SOFTWARE\\Puppet Labs"}
+      let (:subkey_name) { "PuppetRegistryTest" }
+      let (:guid) { SecureRandom.uuid }
+
+      after(:each) do
+        # Ruby 2.1.5 has bugs with deleting registry keys due to using ANSI
+        # character APIs, but passing wide strings to them (facepalm)
+        # https://github.com/ruby/ruby/blob/v2_1_5/ext/win32/lib/win32/registry.rb#L323-L329
+        # therefore, use our own built-in registry helper code
+
+        hklm.open(puppet_key, Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          subject.delete_key(reg, subkey_name)
+        end
       end
 
-      it "passes REG_DWORD through" do
-        reg_value = ['dword', Win32::Registry::REG_DWORD, '1']
-
-        value = expects_registry_value(reg_value)
-
-        expect(Integer(value)).to eq(1)
+      # proof that local encodings (such as IBM437 are no longer relevant)
+      it "will return a UTF-8 string from a REG_SZ registry value (written as UTF-16LE)",
+        :if => RUBY_VERSION >= '2.1' do
```

Comment:
No great way of making this test run correctly on `2.0.0` or `1.9.3` (which would be a temporary measure for the sake of CI), so instead just drop a guard in here to make sure it doesn't cause CI failures.